### ISSUE TITLE
Audio stutter and missing frames with Unity 2022 and new orchestrator

### ISSUE THREAD

**Issue by jackjansen** [cwi-dis/VR2Gather#141]
The problem is most clear when playing Mediascape with headsets, but not limited to that. But it is probably the best testbed.

Since the merge of these two features audio is stuttering. Also, if you look at the rising particle clouds in the training area you'll see that the process is not continuous: every few seconds there will be a break.

Lowering the load (by diminishing point cloud FPS, or lowering the point count) makes the problem less, but not go away.

Our gut feeling (@ashutosh3308 and @jackjansen ) is that the problem is either with the switch to 2022, or with the switch to the new orchestrator, specifically with the switch to the new BestHTTP (because in Mediascape as in production audio and pointclouds are sent over direct TCP, not through the orchestrator).

Possibly some BestHTTP call has become blocking, or we are using a different BestHTTP API call (where we used a non-blocking call before but a blocking call now). @troeggla any ideas?

It also seems that we're seeing garbage collection cycles in the performance monitor every time we see the particles stop moving steadily upward. So maybe we somehow messed up our garbage collection settings or we need a new setting? This would point more towards a Unity 2022 issue.

**Comment by troeggla** [comment]
The only place I can think of that deals with that data is [here](https://github.com/cwi-dis/VR2Gather/commit/4f3c20f45c2837aec7ab13d9277f897620b39a46), where in the new version of BestHTTP I had to actually copy the binary data out of the payload packet and into a new byte array. That might account for an increase in GC cycles.

**Comment by jackjansen** [comment]
@troeggla thanks, that's interesting to know (and to remember). But it's probably not the cause of the current issue, because we are also seeing this issue with TCP transport, so then this code wouldn't be triggered (or at least it wouldn't be triggered for the large volume data).

**Comment by jackjansen** [comment]
Interesting find: Upgraded from Unity 2022.3.10 to 2023.3.20 just now, at home, and the problem seems to be significantly less, possibly even completely gone.

Need to test with vrbig/vrsmall.

**Comment by jackjansen** [comment]
Tested this today at home, in mediascape, between Mac and Windows. Windows both with screen and HMD. Compressed socketIO audio and compressed synthetic pointclouds.

No problems anymore. Audio was fine in both directions, with both HMD and screen.

@ashutosh3308 we need to confirm tomorrow between vrbig and vrsmall.

My guess is that some of the changes I did to the audio pipeline have fixed this, but it might also be some other change over the last weeks.
